crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -1776,20 +1820,20 @@ impl<'db> TypeInferenceBuilder<'db> {
                         asname: _,
                     } = alias;
 
-                    let member_ty = module_ty.member(self.db, &ast::name::Name::new(&name.id));
-
-                    if member_ty.is_unbound() {
+                    if let SymbolLookupResult::Type(member_ty, _) =
+                        module_ty.member(self.db, &ast::name::Name::new(&name.id))
```

Comment:
Nit: Consider adding a `as_type` that returns an `Option<Type<'db>>` to remove the need to ignore the boundness